weather_aplication/ShowWeatherViewController.swift

```diff
@@ -0,0 +1,186 @@
+import SnapKit
+import UIKit
+
+struct WeatherData {
+    let hour: String
+    let temperature: Double
+    let pressure: Double
+    let windSpeed: Double
+    let weatherCode: Int
+}
+
+class ShowWeatherViewController: UIViewController{
+    private let kSideMargin: CGFloat = 10
+    private let kTopMargin: CGFloat = 20
+    private let kBottomMargin: CGFloat = 5
+    private let kElementsToShow: Int = 24
+    private let kHeaderHeight: CGFloat = 50
+    
+    private let longitude: Double
+    private let latitude: Double
+    
+    private let cityLabel = UILabel()
+    private let weatherTableView = UITableView()
+    
+    private var date = String()
+    
+    private var dataArray: [WeatherData] = []
+    private var urlString: String {
+       let url =  "https://api.open-meteo.com/v1/forecast?latitude=\(latitude)&longitude=\(longitude)&hourly=temperature_2m,surface_pressure,weathercode,windspeed_10m"
+        return url
+    }
```

Comment:
moim zdaniem jeżeli tak ważne pole jak urlString opiera się o inne pola klasy to powinno być tworzone inaczej. Masz tylko "bazę" takiego urla i resztę "dopisujesz" przy tworzeniu